build/Microsoft.DotNet.Cli.Prepare.targets

```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Layout" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  
+  <UsingTask TaskName="Rid" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="Architecture" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="GenerateBuildVersionInfo" AssemblyFile="$(CLIBuildDll)" />
+
+  <Target Name="Init" >
+    <!-- Current Runtime Information -->
+    <Rid>
+      <Output TaskParameter="OutputRid" PropertyName="Rid" />
+    </Rid>
+    <Architecture>
+      <Output TaskParameter="OutputArchitecture" PropertyName="Architecture" />
+    </Architecture>
+
+    <!-- Common Properties -->
+    <PropertyGroup>
+      <RepoRoot>$(MSBuildThisFileDirectory)/..</RepoRoot>
+      <BaseOutputDirectory>$(MSBuildThisFileDirectory)/../artifacts/$(Rid)</BaseOutputDirectory>
+      <OutputDirectory>$(BaseOutputDirectory)/stage2</OutputDirectory>
+      <Stage2CompilationDirectory>$(BaseOutputDirectory)/stage2compilation</Stage2CompilationDirectory>
+      <IntermediateDirectory>$(BaseOutputDirectory)/intermediate</IntermediateDirectory>
+      <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    </PropertyGroup>
+
+    <!-- Generates BuildVersion Item
+        BuildVersion Metadata:
+          - Major
+          - Minor
+          - Patch
+          - ReleaseSuffix
+          - CommitHash
+          - CommitCount
+          - VersionSuffix
+          - SimpleVersion
+          - NugetVersion
+          - MsiVersion            -->
+    <GenerateBuildVersionInfo RepoRoot="$(RepoRoot)">
+        <Output TaskParameter="OutputBuildVersionInfo"
```

Comment:
Same comment as https://github.com/dotnet/cli/pull/3696/commits/5ce1fd6f4de0130f4ac62a3daa78838ffdc2aa7d#r68623008.  It seems like we are using an Item here just for grouping multiple properties.  Would it be better if this Task generated a bunch of properties instead of an Item?